examples/src/createReactAgent/accessLongTermMemory.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Access Long Term Memory in Message Generation
+ *
+ * This capability allows the agent to retrieve and incorporate information from
+ * persistent memory stores (like vector databases) when generating responses,
+ * enabling recall of past interactions and learned information.
+ *
+ * Why this is important:
+ * - Knowledge Persistence: Retains and recalls important information across sessions, creating a sense of continuity and learning
+ * - Contextual Enrichment: Enhances responses with relevant historical context and previously discussed topics
+ * - Personalized Experience: Builds long-term understanding of user preferences, needs, and interaction patterns
+ *
+ * Example Scenario:
+ * You're building a personal assistant for a busy executive. When they mention "the Johnson project",
```

Comment:
i think importantly compared to example below, the agent isnt decidding what to look up, we are doing it programmatically. procedural memory (eg parts of the system prompt) may be better example here, rather than semantic search